dev-tools/xtask/src/download.rs

```diff
@@ -0,0 +1,935 @@
+        let commit_path =
+            self.versions_dir.join("maghemite_mg_openapi_version");
+        let [commit] = get_values_from_file(["COMMIT"], &commit_path).await?;
+
+        let repo = "oxidecomputer/maghemite";
+        let base_url = format!("{BUILDOMAT_URL}/{repo}/image/{commit}");
+
+        let filename = "mgd.tar.gz";
+        let tarball_path = download_dir.join(filename);
+        download_file_and_verify(
+            &self.log,
+            &tarball_path,
+            &format!("{base_url}/{filename}"),
+            ChecksumAlgorithm::Sha2,
+            &mgd_sha2,
+        )
+        .await?;
+        unpack_tarball(&self.log, &tarball_path, &download_dir).await?;
+
+        let destination_dir = self.output_dir.join("mgd");
+        let _ = tokio::fs::remove_dir_all(&destination_dir).await;
+        tokio::fs::create_dir_all(&destination_dir).await?;
+        copy_dir_all(
+            &download_dir.join("root"),
+            &destination_dir.join("root"),
+        )?;
+
+        let binary_dir = destination_dir.join("root/opt/oxide/mgd/bin");
+
+        match os_name()? {
+            Os::Linux => {
+                let filename = "mgd";
+                let path = download_dir.join(filename);
+                download_file_and_verify(
+                    &self.log,
+                    &path,
+                    &format!(
+                        "{BUILDOMAT_URL}/{repo}/linux/{commit}/{filename}"
+                    ),
+                    ChecksumAlgorithm::Sha2,
+                    &mgd_linux_sha2,
+                )
+                .await?;
+                set_permissions(&path, 0o755).await?;
+                tokio::fs::copy(path, binary_dir.join(filename)).await?;
+            }
+            _ => (),
+        }
+
+        Ok(())
+    }
+
+    async fn download_softnpu(&self) -> Result<()> {
+        let destination_dir = self.output_dir.join("npuzone");
+        tokio::fs::create_dir_all(&destination_dir).await?;
+
+        let repo = "oxidecomputer/softnpu";
+
+        // TODO: This should probably live in a separate file, but
+        // at the moment we're just building parity with
+        // "ci_download_softnpu_machinery".
+        let commit = "dbab082dfa89da5db5ca2325c257089d2f130092";
+
+        let filename = "npuzone";
+        let base_url = format!("{BUILDOMAT_URL}/{repo}/image/{commit}");
+        let artifact_url = format!("{base_url}/{filename}");
+        let sha2_url = format!("{base_url}/{filename}.sha256.txt");
+        let sha2 = reqwest::get(sha2_url).await?.text().await?;
+        let sha2 = sha2.trim();
+
+        let path = destination_dir.join(filename);
+        download_file_and_verify(
+            &self.log,
+            &path,
+            &artifact_url,
+            ChecksumAlgorithm::Sha2,
+            &sha2,
+        )
+        .await?;
+        set_permissions(&path, 0o755).await?;
+
+        Ok(())
+    }
+
+    async fn download_thundermuffin(&self) -> Result<()> {
```

Comment:
Strictly speaking I don't think we need thundermuffin here. It's needed as an [intermediate package](https://github.com/oxidecomputer/omicron/blob/dbf849401c08c62bfcc64c28292f048469b10ef8/package-manifest.toml#L658-L665) for building the probe zone image. But we can forgo downloading it here I think.

Reply:
I added this to correspond to https://github.com/oxidecomputer/omicron/blob/main/tools/ci_download_thundermuffin , which is currently being invoked by `./tools/install_builder_prerequisites.sh`.

Just to confirm: Are you okay with us not invoking `ci_download_thundermuffin`, and also deleting that bash script?

Reply:
Yup, that should be fine.